R/sim_plot_helpers.R

```diff
@@ -0,0 +1,170 @@
+            title = 'Distribution',
+            labels = c('Normal','Gamma', 'Uniform'),
+            size = 3,
+            title.cex = 1,
+            label.cex = 1,
+            border = 'black'
+            )
+        )
+      )
+    }
+
+  if (! is.null(cont.params.to.include)) {
+    cont.params <- gsub('_', '.', cont.params.to.include)
+
+    cont.params.legend <- lapply(cont.params, function(p) {
+    cont.param.name <- sub('[.]decile', '', p);
+
+      list(
+        colours = c('white', covariate.col[[p]]),
+        labels = if (cont.param.name == 'jaccard') c('0', '1') else as.character(unimodal.params[[cont.param.name]]),
+        at = c(0, 100),
+        title = cont.param.name,
+        angle = -90,
+        width = 6,
+        continuous = TRUE
+        )
+      })
+
+    names(cont.params.legend) <- rep('legend', length(cont.params.legend))
+
+    legend.list <- c(legend.list, cont.params.legend)
+    }
+
+  BoutrosLab.plotting.general::legend.grob(
+    legends = legend.list,
+    label.cex = 1.5,
+    title.cex = 1.5,
+    size = 3,
+    title.just = 'left',
+    layout = c(1,length(legend.list))
+    )
+  }
+
+
+
+assign.cov.factor.col <- function(x, ramp.colors) {
+  x.levels <- levels(x);
+  k <- length(x.levels);
+
+  colormap <- setNames(
+    colorRampPalette(ramp.colors)(k),
+    x.levels
+    );
+
+  colormap[x];
+}
+
+sim.plot.heatmap.cov <- function(cov.data) {
+  if ('max_uniform' %in% colnames(cov.data)) {
+    cov.data$max_uniform <- ifelse(cov.data$max_uniform, covariate.col$max.uniform, 'white');
+  }
+  if ('remove_low_entropy' %in% colnames(cov.data)) {
+    cov.data$remove_low_entropy <- ifelse(cov.data$remove_low_entropy, covariate.col$remove.low.entropy, 'white');
+  }
+  if ('actual_dist' %in% colnames(cov.data)) {
+    cov.data$actual_dist <- unname(distribution_colours[as.character(cov.data$actual_dist)]);
+  }
+  if ('dist' %in% colnames(cov.data)) {
+    # TODO: Combine
+    cov.data$dist <- unname(distribution_colours[as.character(cov.data$dist)]);
+  }
+  if ('jaccard_decile' %in% colnames(cov.data)) {
+    cov.data$jaccard_decile <- assign.cov.factor.col(cov.data$jaccard_decile, c('white', covariate.col$jaccard.decile));
+  }
+  if ('N_decile' %in% colnames(cov.data)) {
+    cov.data$N_decile <- assign.cov.factor.col(cov.data$N_decile, c('white', covariate.col$N.decile));
+  }
+  if ('noise_decile' %in% colnames(cov.data)) {
+    cov.data$noise_decile <- assign.cov.factor.col(cov.data$noise_decile, c('white', covariate.col$noise.decile));
+  }
+  if ('eps_decile' %in% colnames(cov.data)) {
+    cov.data$eps_decile <- assign.cov.factor.col(cov.data$eps_decile, c('white', covariate.col$eps.decile));
+  }
+  if ('param_decile' %in% colnames(cov.data)) {
+    cov.data$param_decile <- assign.cov.factor.col(cov.data$param_decile, c('white', covariate.col$param.decile));
+  }
+
+  if (ncol(cov.data) == 1) cov.data <- cbind(cov.data[, 1], cov.data[, 1]);
+
+  create.heatmap(
+    cov.data,
```

Comment:
interesting! I literally did not know you can assign colours to a matrix and create a heatmap *facepalm* - have been numerically coding my colours for years